rotkehlchen/chain/ethereum/manager.py

```diff
@@ -158,15 +158,23 @@ def is_synchronized(self, current_block: int, latest_block: int) -> Tuple[bool,
     def set_rpc_endpoint(self, endpoint: str) -> Tuple[bool, str]:
         """ Attempts to set the RPC endpoint for the ethereum client.
 
-        Returns a tuple (result, message)
-            - result: Boolean for success or failure of changing the rpc endpoint
-            - message: A message containing information on what happened. Can
-                       be populated both in case of success or failure"""
-        result, message = self.attempt_connect(endpoint)
-        if result:
-            log.info('Setting ETH RPC endpoint', endpoint=endpoint)
-            self.ethrpc_endpoint = endpoint
-        return result, message
+           Returns a tuple (result, message)
+               - result: Boolean for success or failure of changing the rpc endpoint
+               - message: A message containing information on what happened. Can
+                          be populated both in case of success or failure"""
+        if endpoint == '':
+            if self.web3:
+                del self.web3
+                self.web3 = None
```

Comment:
Without setting to None, setting setting a valid connection would fail because of the missing property.

Reply:
yea due to `if self.web3`